docker/archive/dest.go

```diff
@@ -19,15 +19,24 @@ func newImageDestination(ctx *types.SystemContext, ref archiveReference) (types.
 	if ref.destinationRef == nil {
 		return nil, errors.Errorf("docker-archive: destination reference not supplied (must be of form <path>:<reference:tag>)")
 	}
-	fh, err := os.OpenFile(ref.path, os.O_WRONLY|os.O_EXCL|os.O_CREATE, 0644)
+
+	// ref.path can be either a pipe or a regular file
+	// in the case of a pipe, we must be able to open a pre-existing file
```

Comment:
+	// in the case of a pipe, we must be able to open a pre-existing file 
This line is wrong.
+     //in the case of a pipe, we require that we can open it for write